test/functional/test_runner.py

```diff
@@ -264,7 +285,7 @@ def main():
 
     # Remove the test cases that the user has explicitly asked to exclude.
     if args.exclude:
-        exclude_tests = [re.sub("\.py$", "", test) + ".py" for test in args.exclude.split(',')]
+        exclude_tests = [re.sub("\.py$", "", test) + (".py" if ".py" not in test else "") for test in args.exclude.split(',')]
```

Comment:
I don't understand this change. The point of:

```
[re.sub("\.py$", "", test) + ".py"
```
is that if `<test_name>.py` or `<test_name>` is given, then it's normalized to `<test_name>.py`.

Your new version:

```
re.sub("\.py$", "", test) + (".py" if ".py" not in test else "")
```

 essentially toggles, so `<test_name>.py` is changed to `<test_name>` and `<test_name>` is changed to `<test_name>.py`.

```
>>> test1 = 'test_name1.py'
>>> test2 = 'test_name2'
>>> for test in [test1,test2]:
...     print(re.sub("\.py$", "", test) + ".py") # original version
... 
test_name1.py
test_name2.py
>>> for test in [test1,test2]:
...     print(re.sub("\.py$", "", test) + (".py" if ".py" not in test else "")) # new version
... 
test_name1
test_name2.py
```

This breaks using `-exclude=<test_name>.py`

Reply:
The goal was to be able to specify `--exclude "feature_notifications,wallet_multiwallet,wallet_multiwallet.py --usecli"`.

I think removing everything from the `test_list` that `.startswith(item.split('.py')[0])` for any item in `args.exclude.split(',')` should solve all issues?

Reply:
Oh, I didn't test that case. Should be fixed in #14316